src/Symfony/Component/HttpFoundation/UriSigner.php

```diff
@@ -158,4 +173,49 @@ private function getExpirationTime(\DateTimeInterface|\DateInterval|int $expirat
 
         return (string) $expiration;
     }
+
+    /**
+     * @return self::STATUS_*
+     */
+    private function doVerify(string $uri): int
+    {
+        $url = parse_url($uri);
+        $params = [];
+
+        if (isset($url['query'])) {
+            parse_str($url['query'], $params);
+        }
+
+        if (empty($params[$this->hashParameter])) {
+            return self::STATUS_MISSING;
+        }
+
+        $hash = $params[$this->hashParameter];
+        unset($params[$this->hashParameter]);
+
+        // In 8.0, remove support for non-url-safe tokens
```

Comment:
to make it easier to spot when preparing 8.0
```suggestion
        // @deprecated: In 8.0, remove support for non-url-safe tokens
```

Reply:
alternatively, we might just remove the comment as it might break stuff and keeping the strtr doesn't hurt?

Reply:
I've removed the comment.